server/viame_tasks/tasks.py

```diff
@@ -150,3 +167,50 @@ def convert_video(self, path, folderId, token, auxiliaryFolderId):
         _file["itemId"], {"folderId": folderId, "codec": "h264"}
     )
     os.remove(output_path)
+
+
+@app.task(bind=True)
+def convert_images(self, folderId, token):
+    """
+    Ensures that all images in a folder are in a web friendly format (png or jpeg).
+
+    If conversions succeeds for an image, it will replace the image with an image
+    of the same name, but in a web friendly extension.
+
+    Returns the number of images successfully converted.
+    """
+    gc = self.girder_client
+    gc.token = token
+
+    items = gc.listItem(folderId)
+    skip_item = (
+        lambda item: item["name"].endswith(".png")
+        or item["name"].endswith(".jpeg")
+        or item["name"].endswith(".jpg")
+    )
+    valid_items = [item for item in items if not skip_item(item)]
+
+    count = 0
+    with tempfile.TemporaryDirectory() as temp:
+        dest_dir = Path(temp)
+
+        for item in valid_items:
+            # Assumes 1 file per item
+            gc.downloadItem(item["_id"], dest_dir, item["name"])
+
+            item_path = dest_dir / item["name"]
```

Comment:
This file uses both pathlib `/` style and `os.path.join`.  Not sure which is better, but should probably settle on one.

Reply:
It seems that making this change would modify a decent chunk of code, so I'll create an issue for it.